### ISSUE TITLE
Training Fails

### ISSUE THREAD

**Issue by akif-caglar** [bcmi/GracoNet-Object-Placement#7]
Hi, I try to run training code of GracoNet, main.py, on my local computer and want to see if I can get similar results to your pretrained weights. 
However, training seems to fail. In first epochs all objects are placed with 0 scale ratio, so they do not appear on the screen. And at last epochs like around 9, all objects are pasted with maximum scale to same location. 
I have not changed anything from main.py and model.py, only batch_size. Can you help me please ? Are you able to train new model well with the current state of the code which is on github ?  Have you encountered some issue like this ? 
Thank you very much for your time and your work.

**Comment by Siyuan-Zhou** [comment]
@akif-caglar Hi, thanks for your question. Although I did not encounter your described issue before, I will train the model again with the current code on Github. If I encounter your described issue, I will try to find the potential bug.

**Comment by akif-caglar** [comment]
@Siyuan-Zhou  Thank you so much. I am looking forward to hear from you.

**Comment by akif-caglar** [comment]
Hi, its me again. I wanted to give more details about the situation and ask for your opinion. These pictures are from samples. Around iteration 46500 generator starts to collapse on to a state in which it places either very large objects in same place mostly or places with 0 scale so object is not visible.
![image](https://user-images.githubusercontent.com/16181878/206019367-46344ca8-76b4-4fd4-b948-05b86432c8a3.png)
![image](https://user-images.githubusercontent.com/16181878/206019514-0ec0e74c-4a53-434c-9ce0-90a64ee6b195.png)

This collapse also can be seen from losses around iteration 45k. I though it is beacuse performance gap between discriminator and generator becomes too large on that point and from there generator starts to diverge. As a solution right now I am freezing weights of discriminator and also I have increased reconstruction loss's multiplier from 50 to 450 during training. 

I would be happy to hear if you have any advice. Also, have you trained generator and discriminator at the same time during whole training without freezing weights of one? 
![image](https://user-images.githubusercontent.com/16181878/206020093-90523c47-d67c-4d46-992e-67abd1c7f865.png)
![image](https://user-images.githubusercontent.com/16181878/206020117-2fd11a1e-61c1-44d7-827d-a4ecf107ce79.png)

**Comment by Siyuan-Zhou** [comment]
I have trained the model again with batch size 8 and I encountered the same problem as you mentioned. It seems that a larger batch size (like 32) may contribute to more reasonable generation results. Moreover, although the model falls into modal collapse in some epochs, you could also continue the training process. The model may behave well and go out of modal collapse automatically after several epochs. Last but not least, using evaluation metrics (instead of only focusing on losses) to examine the performances is more straightforward. I advise that you could try to print out the evaluation results of each epoch.

**Comment by DavidXie03** [comment]
I am faced with the similar problem. Default parameters were used, but the final result was that the generator placed every foreground object in the left-bottom corner.
![09151af09b5dfc9e1314557fe47308f](https://github.com/bcmi/GracoNet-Object-Placement/assets/62877238/461755df-ea0c-4fb1-9e02-3f36ea5870f7)
![29679ebca34de615dbfcc0809f16b2f](https://github.com/bcmi/GracoNet-Object-Placement/assets/62877238/7151b62b-6b8b-42ee-9ee3-5c9fc729c816)
![aad07f1de0893026a3e2e374c68242a](https://github.com/bcmi/GracoNet-Object-Placement/assets/62877238/e0b06a80-4fd0-41a0-aa52-c7ec6abcf8f3)
![f6f52a0c926a81cee5d3eb468cabe86](https://github.com/bcmi/GracoNet-Object-Placement/assets/62877238/f924e1e1-660d-4545-85a9-56e7495658d8)

**Comment by ustcnewly** [comment]
The adversarial loss may make the training difficulty, you can try replacing adversarial loss with classification loss (using fixed SOPA as binary classification model). Additionally, I recommend FOPA https://github.com/bcmi/FOPA-Fast-Object-Placement-Assessment, which is more stable and effective.
